packages/core/src/util/transactionSummaryInspector.ts

```diff
@@ -0,0 +1,242 @@
+import * as Cardano from '../Cardano';
+import { AssetId } from '../Cardano';
+import {
+  AssetsMintedInspection,
+  Inspector,
+  ResolutionResult,
+  assetsBurnedInspector,
+  assetsMintedInspector,
+  resolveInputs,
+  totalAddressInputsValueInspector,
+  totalAddressOutputsValueInspector
+} from './txInspector';
+import { BigIntMath } from '@cardano-sdk/util';
+import { addTokenMaps, subtractTokenMaps } from '../Asset/util';
+import { coalesceValueQuantities } from './coalesceValueQuantities';
+import { subtractValueQuantities } from './subtractValueQuantities';
+
+interface TransactionSummaryInspectorArgs {
+  addresses: Cardano.PaymentAddress[];
+  rewardAccounts: Cardano.RewardAccount[];
+  inputResolver: Cardano.InputResolver;
+  protocolParameters: Cardano.ProtocolParameters;
+}
+
+export type TransactionSummaryInspection = {
+  tokens: Cardano.TokenMap;
+  coins: Cardano.Lovelace;
+  collateral: Cardano.Lovelace;
+  deposit: Cardano.Lovelace;
+  returnDeposit: Cardano.Lovelace;
+  totalFees: Cardano.Lovelace;
```

Comment:
Name implies that there is more than 1 type of fee accounted for here, which is not true. If we ever need to find other types of fees (e.g. dex fees), then those would probably be listed in a new property or separate inspectors to have more detail.
```suggestion
  fee: Cardano.Lovelace;
```

Reply:
done here https://github.com/input-output-hk/cardano-js-sdk/pull/1057/commits/289d353cfccce8f9b1f5d6871367e9234a559bce (and ifxed a few bugs with the type as well)